src/repose.c

```diff
@@ -101,6 +102,15 @@ static _noreturn_ void elephant(void)
 
 static inline int clone_pkg(const struct repo *repo, const struct pkg *pkg)
 {
+    _cleanup_free_ char *signame = joinstring(pkg->filename, ".sig", NULL);
+    _cleanup_close_ int sigsrc = openat(repo->poolfd, signame, O_RDONLY);
+    _cleanup_close_ int sigdest = openat(repo->rootfd, signame, O_WRONLY | O_TRUNC, 0664);
+
+    if (sigdest < 0 && errno == ENOENT && sigsrc > -1)
```

Comment:
Am I missing something or will this condition always be false? Since `sigdest` is opened last, if `sigdest` > 0, `errno` should be 0, right?